examples/all-clusters-app/linux/main.cpp

```diff
@@ -68,6 +74,34 @@ static Identify gIdentify1 = {
     chip::EndpointId{ 1 }, OnIdentifyStart, OnIdentifyStop, EMBER_ZCL_IDENTIFY_IDENTIFY_TYPE_VISIBLE_LED, OnTriggerEffect,
 };
 
+// Network commissioning
+namespace {
+// This file is being used by platforms other than Linux, so we need this check to disable related features since we only
+// implemented them on linux.
+#if CHIP_DEVICE_LAYER_TARGET_LINUX
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+NetworkCommissioning::LinuxThreadDriver sLinuxThreadDriver;
+Clusters::NetworkCommissioning::Instance sThreadNetworkCommissioningInstance(0 /* Endpoint Id */, &sLinuxThreadDriver);
+#endif
+#if CHIP_DEVICE_CONFIG_ENABLE_WPA
+NetworkCommissioning::LinuxWiFiDriver sLinuxWiFiDriver;
+Clusters::NetworkCommissioning::Instance sWiFiNetworkCommissioningInstance(1 /* Endpoint Id */, &sLinuxWiFiDriver);
```

Comment:
From a client perspective, how do we discover what endpoints exist and whether they have a network commissioning cluster?

Reply:
There are a few possibilities, IIRC, there is a descriptor cluster for listing all clusters exists on the device.

However, it is always possible to send a wildcard read request for reading feature map attribute of network commissioning cluster on all endpoints.

In chip-repl, we can do:
`await devCtrl.ReadAttribute(nodeid=1, attributes=[(Clusters.NetworkCommissioning.Attributes.FeatureMap)])`

You will get something like this:

```
{
  0: [NetworkCommissioning.Attributes.FeatureMap: 2],
  1: [NetworkCommissioning.Attributes.FeatureMap: 1]
}
```

You can get the list of endpoints as well as the interface type.

Reply:
e.g.

```
In [2]: await devCtrl.ReadAttribute(nodeid=1, attributes=[(Clusters.NetworkCommissioning.Attributes.FeatureMap)])
Out[2]: 
{
│   0: {
│   │   <class 'chip.clusters.Objects.NetworkCommissioning'>: {
│   │   │   <class 'chip.clusters.Objects.NetworkCommissioning.Attributes.FeatureMap'>: 2
│   │   }
│   },
│   1: {
│   │   <class 'chip.clusters.Objects.NetworkCommissioning'>: {
│   │   │   <class 'chip.clusters.Objects.NetworkCommissioning.Attributes.FeatureMap'>: 1
│   │   }
│   }
}
```

I will update the script in this PR to make use of this feature.

Reply:
Updated test script